### PR TITLE
Fix several IO tests and remove some stale ActiveIssue

### LINKED ISSUE #6153: System.IO.Tests.FileStream_Dispose.DisposeVirtualBehavior failed on Ubuntu Release
```
 System.IO.Tests.FileStream_Dispose.DisposeVirtualBehavior [FAIL]
17:42:30       Assert.True() Failure
17:42:30       Expected: True
17:42:30       Actual:   False
17:42:30       Stack Trace:
17:42:31             at System.IO.Tests.FileStream_Dispose.DisposeVirtualBehavior()
```

http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/ubuntu_release_tst_prtest/1916/consoleFull#-21371504001f1a4601-6aec-4fd5-b678-78d4389fd5e8

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @ianhays 
Fixes #6153

### CODE REVIEW

File: src/System.IO.FileSystem/tests/FileStream/Dispose.cs [hunk 1]
```diff
@@ -58,49 +58,67 @@ protected override void Dispose(bool disposing)
...
-            called = false;
-            MyFileStream fs2 = new MyFileStream(GetTestFilePath(), FileMode.Create);
-            fs2.DisposeMethod = (disposing) =>
+            Action act = () => // separate method to avoid JIT lifetime-extension issues
             {
-                called = true;
-                Assert.False(disposing);
+                var fs2 = new MyFileStream(GetTestFilePath(), FileMode.Create)
+                {
+                    DisposeMethod = (disposing) =>
```
Comment (ianhays) [inline src/System.IO.FileSystem/tests/FileStream/Dispose.cs]: `fs2.DisposeMethod = ...`?

Comment (stephentoub) [inline src/System.IO.FileSystem/tests/FileStream/Dispose.cs]: This is an object initializer. https://msdn.microsoft.com/en-us/library/bb384062.aspx

**Comment by ianhays** [comment]
LGTM minus the one question. Thanks for figuring this one out Steve! The lifetime extension stumped me.

File: src/System.IO.FileSystem/tests/FileStream/Dispose.cs (hunk 1 above)
Comment (ianhays) [inline src/System.IO.FileSystem/tests/FileStream/Dispose.cs]: Ah, I see the semicolon now. I was comparing it to `Dispose_CallsVirtualDispose_TrueArg` where the DisposeMethod setting is within a `using`.

Comment (stephentoub) [inline src/System.IO.FileSystem/tests/FileStream/Dispose.cs]: Yeah, in the other test, it looked weird using a multi-line object initializer in the using, so I left that one as a normal property access.

**Comment by stephentoub** [comment]
Thanks, Ian.

**Pull request merged**